### DISCUSSION TITLE
Kotlin-like scoping functions

### DISCUSSION THREAD

**Discussion by vlad-iancu** [dotnet/csharplang#4551] [Language Ideas]
I find it really useful if C# would have an equivalent for Kotlin `also`, `let`, `apply` and `run` functions

**Comment by HaloFour** [comment]
I think the features described by this discussion would enable writing such functions: https://github.com/dotnet/csharplang/discussions/1151

**Comment by m00shm00sh** [comment]
`let` and `also` can be simulated with generic extension methods but apply and run would require lambdas with receivers, which requires language changes
